rpa-worker-application/src/functionalTest/groovy/io/camunda/rpa/worker/files/api/FilesFunctionalSpec.groovy

```diff
@@ -246,4 +198,90 @@ Do Nothing
 		resp['main.robot'].metadata().fileName() == "main.robot"
 		resp.values()*.contentHash().every { it == "content-hash" }
 	}
+
+	private static final String UPLOAD_FILES_WITH_UNNORMALISED_PATHS_SCRIPT = '''\
+*** Settings ***
+Library    Camunda
+Library    OperatingSystem
+
+*** Tasks ***
+Test
+    Create File    one.txt
+    Create File    two/two.txt
+    Create File    two/three.txt
+    Create directory    two/four
+    Upload Documents    ./one.txt    uploaded1
+    Upload Documents    two/four/../two.txt    uploaded2
+    Upload Documents    ./tw*/four/../*.txt    uploaded3
+'''
+
+	@Issue("https://github.com/camunda/rpa-worker/issues/129")
+	void "Correctly handles upload file requests with unnormalised paths"() {
+		given:
+		bypassZeebeAuth()
+		Workspace theWorkspace
+		workspaceCleanupService.preserveLast(_) >> { Workspace w ->
+			theWorkspace = w
+			return Mono.empty()
+		}
+
+		and:
+		zeebeApi.dispatcher = this.&copyInputToOutputDispatcher
+
+		when:
+		EvaluateScriptResponse response = post()
+				.uri("/script/evaluate")
+				.body(BodyInserters.fromValue(new EvaluateScriptRequest(UPLOAD_FILES_WITH_UNNORMALISED_PATHS_SCRIPT, [:])))
+				.retrieve()
+				.bodyToMono(EvaluateScriptResponse)
+				.block(Duration.ofSeconds(10))
```

Comment:
This increased timeout will not be necessary, the delay is caused by the way hostname resolution works in Python on Windows. **camunda-utils** is going to change the default Worker URL to use an IP address instead so this request can use the normal timeout mechanism once the lib is updated